### PR TITLE
DTSPO-6489 - fluxv2 ithc migration - neuvector - patch 5

### PR DESCRIPTION
**Pull request by endakelly**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-6489

### Change description ###
Update neuvector.yaml to use flux v2 config

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: apps/neuvector/neuvector/neuvector.yaml [hunk 1]
```diff
@@ -4,28 +4,19 @@ kind: HelmRelease
...
     config:
       autoScan: true
     rules:
       admission:
         enable: true
-        mode: "protect"  # one of: monitor, protect
+        mode: "protect" # one of: monitor, protect
         rules:
           - '{"config":{"id":1000,"category":"Kubernetes","comment":"Allow system namespaces","criteria":[{"name":"namespace","op":"containsAny","value":"neuvector,kube-system,knode-system"}],"disable":false,"rule_type": "exception","cfg_type":"user_created"}}'
-          - '{"config":{"id":1001,"category":"Kubernetes","comment":"Allow admin namespaces","criteria":[{"name":"namespace","op":"containsAny","value":"admin,kured,monitoring,flux-system,dynatrace"}],"disable":false,"rule_type": "exception","cfg_type":"user_created"}}'
```
[the remark this answers is not in the document]
Comment (endakelly) [reply apps/neuvector/neuvector/neuvector.yaml:28]: Mistake. Didn't catch that was different. Fixed now

File: apps/neuvector/neuvector/neuvector.yaml [hunk 2]
```diff
@@ -77,34 +70,6 @@ spec:
...
-                  topologyKey: "kubernetes.io/hostname"
-          nodeAffinity:
-            preferredDuringSchedulingIgnoredDuringExecution:
-              - weight: 1
-                preference:
-                  matchExpressions:
-                    - key: kubernetes.azure.com/mode
-                      operator: In
-                      values:
-                        - system
```
Comment (timja) [inline apps/neuvector/neuvector/neuvector.yaml:107]: what's with all these values getting removed throughout the file?

Comment (endakelly) [reply apps/neuvector/neuvector/neuvector.yaml:107]: Was getting errors around the syntax in the file which was still following a fluxv1 config as far as I could tell so I checked out the config in cft and used that as a guide. I can see from the neuvector-helm repo that the tolerations and affinity values are valid so I can add these back in if you think they need to be there for sds.

Comment (timja) [reply apps/neuvector/neuvector/neuvector.yaml:107]: yes they need to be there for SDS

File: apps/neuvector/neuvector/neuvector.yaml [hunk 3]
```diff
@@ -68,7 +59,9 @@ spec:
       - '{"config":{"name":"Mta","criteria":[{"key": "address", "value": "mta.reform.hmcts.net", "op": "="}]}}'
       - '{"config":{"name":"Idam","criteria":[{"key": "address", "value": "idam-api.platform.hmcts.net", "op": "="}]}}'
     neuvector:
-      tag: 4.3.2
+      tag: 4.2.0
```
Comment (timja) [inline apps/neuvector/neuvector/neuvector.yaml:62]: ? shouldn't be downgrading

File: apps/neuvector/neuvector/neuvector.yaml [hunk 4]
```diff
@@ -68,7 +59,9 @@ spec:
...
+      containerd:
+        enabled: true
```
Comment (timja) [inline apps/neuvector/neuvector/neuvector.yaml:64]: no need to add this its the default in the chart:
https://github.com/hmcts/chart-neuvector/blob/master/neuvector-azure-keyvault/values.yaml#L30-L31

**Review by timja — APPROVED** [review]

**Pull request merged**